### ISSUE TITLE
[FEATURE] Train hazard model on real historical disruption data and expose model metrics

### ISSUE THREAD

**Issue by 404RiyaFound** [404RiyaFound/NavNER-AI#7]
### Is your feature request related to a problem?

The Stage 2 "AI risk model" cannot learn anything beyond a formula that was hand-written in the same file.

`backend/app/risk_engine.py:41-110` generates 2000 synthetic samples with `np.random.uniform`, computes labels from a fixed linear expression:

```python
landslide_raw = (0.30 * (slope / 60) + 0.28 * (rainfall_24h / 300)
                 + 0.22 * (soil_sat / 100) + 0.12 * base_susceptibility ...)
```

...then fits a `RandomForestClassifier` to those labels at module import time. The model's only achievable outcome is to approximate the equation it was fed. The `RandomForest` adds inference cost and the appearance of ML while contributing zero information the four hardcoded coefficients don't already contain.

**This directly contradicts the Stage 2 PRD**, §2.3 of which specifies a required third data feed:

> **Historical Disruption Baseline:** Ground-truth records of prior landslide and flash-flood coordinates across Assam, Meghalaya, Sikkim, Arunachal Pradesh, Mizoram, Nagaland, and Tripura.

No such dataset is ingested anywhere. There is also no train/test split, no accuracy/precision/recall/AUC measurement, no calibration check, and no model versioning — so nobody can state how well the predictions perform, which is the first question any evaluator or government stakeholder will ask.

### Proposed Solution

1. **Ingest a real historical disruption baseline.** Candidate sources: NASA Global Landslide Catalog (has NER records), GSI landslide inventory, IMD historical rainfall, Assam State Disaster Management Authority flood records, plus the platform's own accumulating verified `incidents` table. Store as a `historical_disruptions` table (location, date, type, severity, source).
2. **Build a real training set.** Join each historical event to the weather conditions *preceding* it (this is why the `weather_telemetry` time-series matters) and to terrain features at that cell. Sample negatives from cells/times with comparable rainfall that did **not** fail — the hard-negative case is what makes the model useful rather than a rainfall threshold in disguise.
3. **Train offline, not at import.** Move training to a script (`backend/scripts/train_risk_model.py`), persist the fitted pipeline with `joblib` (already an installed dependency), and load the artefact at startup. This also removes multi-second import-time training from application boot.
4. **Report metrics.** Add `GET /api/v1/analytics/model-info` returning model version, training date, sample count, and held-out precision/recall/AUC plus a calibration curve. Keep a temporal split (train on earlier monsoons, test on later) — random splits leak badly on spatiotemporal data.
5. **Keep the formula as a documented fallback** for cells with insufficient history, and label predictions with their provenance (`model` vs `heuristic`) so the UI can show confidence honestly.

### Alternatives Considered

- **Keep synthetic training but be honest in the docs** — cheapest fix and better than the current implied claim, but leaves the platform unable to make any validated prediction.
- **Drop the ML layer and ship the formula as an explicit weighted index** — genuinely defensible engineering (it's what the code effectively does today, minus the overhead) but forfeits the AI/ML requirement that is central to the problem statement.

### Additional Context

Recommend also seeding the `landslide_susceptibility_base` values from the published GSI landslide susceptibility zonation maps rather than the hand-assigned constants in `seed.py`.
